0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,387 +110,380 @@ informative:
+  
+## Crypto-conditions as a trigger in distributed systems
 
-## Binary Encoding
+One of the challenges facing a distributed system is achieving atomic execution of a transaction across the system. A common pattern for solving this problem is two-phase commit in which the most time and resource-consuming aspects of the transaction are prepared by all participants following which a simple trigger is sufficient to either commit or abort the transaction. Described in more abstract terms, the system consists of a number of participants that have prepared a transaction pending the fulfillment of a predefined condition.
 
-A description of crypto-conditions is provided in this document using Abstract Syntax Notation One (ASN.1) as defined in [ITU.X680.2015](#itu.X680.2015). Implementations of this spec MUST support encoding and decoding using Octet Encoding Rules (OER) as defined in [ITU.X696.2015](#itu.X696.2015).
+Crypto-conditions defines a mechanism for expressing these triggers as pairs of unique trigger identifiers (conditions) and cryptographically verifiable triggers (fulfillments) that can be deterministically verified by all participants.
 
-## String Types
+It is also important that all participants in such a distributed system are able to evaluate, prior to the trigger being fired, that they will be capable of verifying the trigger. Determinism is useless if validation of the trigger requires algorithms or resources that are not available to all participants.
 
-Crypto-conditions use the following types within string encoding:
+Therefor conditions may be used as **distributable event descriptions** in the form of a *fingerprint*, but also *event meta-data* that allows the event verification system to determine if they have the necessary capabilities (such as required crypto-algorithms) and resources (such as heap size or memory) to verify the event notification later.
 
-BASE10
-: Variable-length integer encoded as a base-10 (decimal) number. Implementations MUST reject encoded values that are too large for them to parse. Implementations MUST be tested for overflows.
+Fulfillments are therefor **cryptographically verifiable event notifications** can be used to verify that the event occurred but also matches the given description.
 
-BASE16
-: Variable-length integer encoded as a base-16 (hexadecimal) number. Implementations MUST reject encoded values that are too large for them to parse. Implementations MUST be tested for overflows. No leading zeros.
+When using crypto-conditions as a trigger it will often make sense for the message that is used for validation to be empty to match the signature of the trigger processing system's API. This makes crypto-conditions compatible with systems that use simple hash-locks as triggers.
```

Comment:
> This helps avoid accidentally created conditions that can't be fulfilled because they require a message to fulfill, but one isn't provided.

Not following this - there is no such thing as a condition that requires a message to fulfill. A condition is just a public key, so given a `condition`, for any possible message `message`, there is a `fulfillment` that fulfills the condition with respect to that message.

A condition is like a public key and a public key can sign any message. The fact that our triggers are signing the empty message is a bit of an arbitrary decision. I can easily imagine someone else's implementation of triggers using crypto-conditions whether the message signed is some constant string defined by the protocol instead.

Sure you could have a public key type which says "this public key can only be used to sign the empty message", but that seems awfully specific and I'm not sure what the benefit of having this feature would be.

Triggers are just one of many ways that you can use a signature scheme and crypto-conditions aren't special compared to other signature schemes in that you can use them as triggers, so I don't see a need to add a special feature for triggers any more than Ed25519 or RSA or any other scheme has a special key format for triggers.